wgengine/router/ifconfig_windows.go

```diff
@@ -576,7 +578,18 @@ func excludeIPv6LinkLocal(in []*net.IPNet) (out []*net.IPNet) {
 	out = in[:0]
 	for _, n := range in {
 		if len(n.IP) == 16 && n.IP.IsLinkLocalUnicast() {
-			continue
+			// Windows creates a fixed link-local address for wintun,
+			// which doesn't seem to route correctly. Unfortunately, LLMNR returns this
+			// address for lookups by the hostname, and Windows prefers using it.
+			// This means that local traffic addressed to the machine's hostname breaks.
+			//
+			// While we otherwise preserve link-local addresses, we delete
+			// this one to force lookups to use a working address.
+			//
+			// See: https://github.com/tailscale/tailscale/issues/4647
+			if isWintun := n.Contains(wintunLinkLocal); !isWintun {
```

Comment:
The netaddr package provides an easy path to equality:

```
winTunLinkLocal = netaddr.MustParseIP(...)

netaddr.FromStdIP(n.IP) == winTunLinkLocal
```

Reply:
Its weird to me that the route is registered as a `/64`. Would that mean that for `fe80::99d0:ec2d:b2e7`, it would get any link-local address? Which I guess implies that my original `.Contains()` would delete every link-local address on the interface.

Maybe this is just a quirk of the API: as long as the IP is `fe80::99d0:ec2d:b2e7` from Windows, the subnet mask shouldn't matter. Do you recall if the API gave us that or just `fe80::/64`?